Core/include/Acts/Clusterization/Clusterization.ipp

```diff
@@ -110,60 +118,69 @@ struct Connections<2> : public ConnectionsBase<4> {
 
 // Cell collection logic
 template <typename Cell, typename Connect, std::size_t GridDim>
-Connections<GridDim> getConnections(typename std::vector<Cell>::iterator it,
-                                    std::vector<Cell>& set, Connect connect) {
+Connections<GridDim> getConnections(std::size_t idx, std::vector<Cell>& cells,
+                                    std::vector<Label>& labels,
+                                    Connect&& connect) {
   Connections<GridDim> seen;
-  typename std::vector<Cell>::iterator it_2{it};
 
-  while (it_2 != set.begin()) {
-    it_2 = std::prev(it_2);
+  for (std::size_t i(0ul); i < idx; ++i) {
```

Comment:
```suggestion
  for (std::size_t i = 0; i < idx; ++i) {
```
nitpick: I think this is more in line what we usually have in ACTS

same below